### PR TITLE
Added new object types and test

### PR DESCRIPTION
**Pull request by petabyte**
Added new object types and test

### CODE REVIEW

File: src/main/java/org/imsglobal/lti2/objects/provider/ToolProfile.java [hunk 1]
```diff
@@ -30,18 +30,21 @@
 
     @JsonProperty("@id")
     @org.codehaus.jackson.annotate.JsonProperty("@id")
-    private String id;
+    private String _id;
```
Comment (bluej100) [inline src/main/java/org/imsglobal/lti2/objects/provider/ToolProfile.java]: Why the change to _id?

File: src/main/java/org/imsglobal/lti2/objects/provider/ToolProfile.java [hunk 2]
```diff
@@ -30,18 +30,21 @@
...
+    private List<ResourceHandler> resource_handler;
+    private List<Message> message;
 
     public ToolProfile() {
     }
 
-    public ToolProfile(String id, String lti_version, JsonNode base_url_choice, ProductInstance product_instance, JsonNode resource_handler, JsonNode message) {
-        this.id = id;
+    public ToolProfile(String _type, String _id, String lti_version, JsonNode base_url_choice, ProductInstance product_instance, List<ResourceHandler> resource_handler, List<Message> message) {
+        this._id = _id;
```
Comment (bluej100) [inline src/main/java/org/imsglobal/lti2/objects/provider/ToolProfile.java]: I imagine you also want to set type here.

File: src/main/java/org/imsglobal/lti2/objects/provider/ToolProfile.java (hunk 1 above)
Comment (petabyte) [inline src/main/java/org/imsglobal/lti2/objects/provider/ToolProfile.java]: Just making it consistent with the ToolConsumerProfile variable definitions
